concrete-core/src/backends/core/implementation/engines/lwe_ciphertext_vector_discarding_addition.rs

```diff
@@ -0,0 +1,160 @@
+use crate::backends::core::implementation::engines::CoreEngine;
+use crate::backends::core::implementation::entities::{
+    LweCiphertextVector32, LweCiphertextVector64,
+};
+use crate::backends::core::private::math::tensor::AsMutTensor;
+use crate::specification::engines::{
+    LweCiphertextVectorDiscardingAdditionEngine, LweCiphertextVectorDiscardingAdditionError,
+};
+
+/// # Description:
+/// Implementation of [`LweCiphertextVectorDiscardingAdditionEngine`] for [`CoreEngine`]
+/// that operates on 32 bits integers.
+impl LweCiphertextVectorDiscardingAdditionEngine<LweCiphertextVector32, LweCiphertextVector32>
+    for CoreEngine
+{
+    /// # Example:
+    /// ```
+    /// use concrete_commons::dispersion::Variance;
+    /// use concrete_commons::parameters::LweDimension;
+    /// use concrete_core::prelude::*;
+    /// # use std::error::Error;
+    ///
+    /// # fn main() -> Result<(), Box<dyn Error>> {
+    /// // DISCLAIMER: the parameters used here are only for test purpose, and are not secure.
+    /// let lwe_dimension = LweDimension(2);
+    /// // Here a hard-set encoding is applied (shift by 20 bits)
+    /// let input_vector = vec![3_u32 << 20; 8];
+    /// let weights_input = vec![2_u32; 8];
+    /// let bias_input = 8_u32 << 20;
+    /// let noise = Variance::from_variance(2_f64.powf(-25.));
+    ///
+    /// let mut engine = CoreEngine::new()?;
+    /// let key: LweSecretKey32 = engine.create_lwe_secret_key(lwe_dimension)?;
+    /// let plaintext_vector: PlaintextVector32 = engine.create_plaintext_vector(&input_vector)?;
+    /// let ciphertext_vector = engine.encrypt_lwe_ciphertext_vector(&key, &plaintext_vector, noise)?;
+    /// let mut output_ciphertext =
```

Comment:
It would be an `output_ciphertext_vector` instead (don't forget to change the destroy as well)